app/src/main/java/com/asfoundation/wallet/ui/widget/holder/TransactionHolder.java

```diff
@@ -169,6 +170,12 @@ private void fill(String from, String to, String currencySymbol, String valueStr
       } else if (transaction.getType()
           .equals(Transaction.TransactionType.IAP_REVERT)) {
         address.setText(R.string.transaction_type_reverted_purchase_title);
+      } else if (transaction.getType()
+          .equals(Transaction.TransactionType.VOUCHER)) {
+        String addressText =
+            details.getSourceName() == null ? isSent ? to : from : getSourceText(transaction);
+        addressText = getString(R.string.voucher_buy_short_header) + " " + addressText;
```

Comment:
I'm not sure, while I don't disagree that it will lead to repeated code, first you would have to get context or an helper class with context to the mapper to use `getString(R.string...)`, and then if you actually need the original sourceName for some reason, you would have to remove the added "Voucher:" which would be pretty funky since you would have to do getString(R.string...) again to remove it.